src/common/hist_util.cu

```diff
@@ -148,11 +148,11 @@ struct GPUSketcher {
    public:
     DeviceShard(int device,
```

Comment:
This DeviceShard class is now redundant right?

Reply:
Yes there are 4-5 places we still have a `DeviceShard` that's no longer necessary. I'll send out a followup PR to clean them up.